src/logging.c

```diff
@@ -116,9 +98,27 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
     g_rcutils_logging_output_handler = &rcutils_logging_console_output_handler;
     g_rcutils_logging_default_logger_level = RCUTILS_DEFAULT_LOGGER_DEFAULT_LEVEL;
 
+    // Check the environment variable for line buffered output
+    const char * line_buffered;
+    const char * ret_str = rcutils_get_env("RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED", &line_buffered);
+
+    if (NULL == ret_str) {
+      if (!strcmp(line_buffered, "1")) {
```

Comment:
An explicit comparison of the return value with a numeric (`== 0`) would be more readable imo and likely avoid the logic error from before to be repeated.

Similar below (`!= 0`).